### PR TITLE
Parameterizing Painless script literals

### PR DESCRIPTION
**Pull request by ycombinator**
Several Filebeat CI runs are failing like so:

```
05:55:30 ======================================================================
05:55:30 FAIL: test_fileset_file_7_traefik (test_modules.Test)
05:55:30 ----------------------------------------------------------------------
05:55:30 Traceback (most recent call last):
05:55:30   File "/go/src/github.com/elastic/beats/filebeat/build/python-env/local/lib/python2.7/site-packages/parameterized/parameterized.py", line 392, in standalone_func
05:55:30     return func(*(a + p.args), **p.kwargs)
05:55:30   File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_modules.py", line 91, in test_fileset_file
05:55:30     cfgfile=cfgfile)
05:55:30   File "/go/src/github.com/elastic/beats/filebeat/tests/system/test_modules.py", line 140, in run_on_file
05:55:30     obj)
05:55:30 AssertionError: not error expected but got: {u'http': {u'version': u'1.0', u'request': {u'method': u'GET'}, u'response': {u'status_code': 200}}, u'log': {u'file': {u'path': u'/go/src/github.com/elastic/beats/filebeat/module/traefik/access/test/test.log'}, u'offset': 1581}, u'url': {u'original': u'/apache_pb.gif'}, u'traefik': {u'access': {u'body_sent': {u'bytes': 2326}, u'user_identifier': u'-'}}, u'@timestamp': u'2000-10-10T20:55:36.000Z', u'read_timestamp': u'2018-12-22T13:50:36.804Z', u'agent': {u'hostname': u'47c93834e587', u'type': u'filebeat', u'version': u'7.0.0'}, u'source': {u'ip': u'127.0.0.1', u'address': u'127.0.0.1'}, u'host': {u'name': u'47c93834e587'}, u'user': {u'name': u'frank'}, u'error': {u'message': u'[script] Too many dynamic script compilations within, max: [75/5m]; please use indexed, or scripts with parameters instead; this limit can be changed by the [script.max_compilations_rate] setting'}, u'input': {u'type': u'log'}, u'event': {u'module': u'traefik', u'dataset': u'access'}}
05:55:30 -------------------- >> begin captured stdout << ---------------------
05:55:30 Using elasticsearch: http://elasticsearch:9200
05:55:30 Testing traefik/access on /go/src/github.com/elastic/beats/filebeat/tests/system/../../module/traefik/access/test/test.log
05:55:30 
05:55:30 --------------------- >> end captured stdout << ----------------------
05:55:30 
05:55:30 ----------------------------------------------------------------------
```

For example:
* https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/2911/beat=filebeat,label=ubuntu/console
* https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/2912/beat=filebeat,label=ubuntu/console

This PR tries to parameterize any painless scripts in Filebeat modules' pipelines to see if it helps with the `Too many dynamic script compilations within, max: [75/5m]` error. Specifically, it parameterizes literals that were previous used in painless scripts by the following Filebeat module pipelines:

* `auditd`
* `nginx`
* `redis`

### CODE REVIEW

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ruflin** [comment]
@ycombinator Thanks for starting to tackle this. I also made some notes about this here: https://github.com/elastic/beats/pull/9777 (see the comments below).

To have much nicer PR's we should probably take on https://github.com/elastic/beats/pull/5339 again. This would make things nicer.

**Comment by ycombinator** [comment]
jenkins test this

File: filebeat/module/auditd/log/ingest/pipeline.json [hunk 1]
```diff
@@ -71,15 +71,18 @@
...
+                "params": {
+                    "single_quote": "'",
+                    "double_quote": "\""
+                }
```
Comment (ycombinator) [inline filebeat/module/auditd/log/ingest/pipeline.json]: This change is currently blocked on https://github.com/elastic/elasticsearch/issues/36993.

Comment (ycombinator) [reply filebeat/module/auditd/log/ingest/pipeline.json]: Worked around this possible ES limitation in subsequent commits.

**Comment by ruflin** [comment]
CI failure does not seem to be related. Need to check if this sneaked into master in an other PR as it complains about traefik and ingesting data into an alias.

File: filebeat/module/auditd/log/ingest/pipeline.json (hunk 1 above)
Comment (ruflin) [reply filebeat/module/auditd/log/ingest/pipeline.json]: Even if we get a fix in ES we probably have to keep the old version around for BC compatibility?

**Comment by ruflin** [comment]
Rebasing on master should get CI green for this one.

**Comment by ycombinator** [comment]
@ruflin CI is green now. When you get a chance, could you review please?

File: filebeat/module/auditd/log/ingest/pipeline.json [hunk 2]
```diff
@@ -71,15 +71,18 @@
...
             "convert": {
-                "field" : "auditd.log.sequence",
+                "field": "auditd.log.sequence",
                 "type": "integer",
                 "ignore_missing": true
             }
         },
         {
             "script": {
-                "lang": "painless",
```
Comment (ruflin) [inline filebeat/module/auditd/log/ingest/pipeline.json]: It seems you removed  `lang: painless`. Was this on purpose?

File: filebeat/module/redis/log/ingest/pipeline.json [hunk 3]
```diff
@@ -1,51 +1,81 @@
...
-      "grok": {
-        "field": "message",
-        "patterns": [
-          "(%{POSINT:process.pid:long}:%{CHAR:redis.log.role} )?%{REDISTIMESTAMP:redis.log.timestamp} %{REDISLEVEL:log.level} %{GREEDYDATA:message}",
-          "%{POSINT:process.pid:long}:signal-handler \\(%{POSINT:redis.log.timestamp}\\) %{GREEDYDATA:message}"
-        ],
-        "pattern_definitions": {
-          "CHAR": "[a-zA-Z]",
-          "REDISLEVEL": "[.\\-*#]"
+    "description": "Pipeline for parsing redis logs",
```
Comment (ruflin) [inline filebeat/module/redis/log/ingest/pipeline.json]: I assume your editor applied here `.editorconfig` which is good: https://github.com/elastic/beats/blob/master/.editorconfig I wonder why this didn't happen for the nginx pipeline.

File: filebeat/module/redis/log/ingest/pipeline.json [hunk 4]
```diff
@@ -1,51 +1,81 @@
...
+                    "warning": "warning"
+                }
+            }
+        },
+        {
+            "script": {
+                "lang": "painless",
+                "source": "if (ctx.redis.log.role == params.master_abbrev) {\n          ctx.redis.log.role = params.master;\n        } else if (ctx.redis.log.role == params.slave_abbrev) {\n          ctx.redis.log.role = params.slave;\n        } else if (ctx.redis.log.role == params.child_abbrev) {\n          ctx.redis.log.role = params.child;\n        } else if (ctx.redis.log.role == params.sentinel_abbrev) {\n          ctx.redis.log.role = params.sentinel;\n        }\n        ",
+                "params": {
+                    "master_abbrev": "M",
```
Comment (ruflin) [inline filebeat/module/redis/log/ingest/pipeline.json]: It seems odd that we have to extract all these constants which are only used once but I don't think there is a better way. Perhaps @jakelandis can comment?

**Comment by ruflin** [comment]
I created a meta issue here to track this: https://github.com/elastic/beats/issues/9821

File: filebeat/module/auditd/log/ingest/pipeline.json (hunk 2 above)
Comment (ycombinator) [reply filebeat/module/auditd/log/ingest/pipeline.json]: Painless is the default language so this field is unnecessary. However, better to be explicit so I'll add it back now.

**Comment by ycombinator** [comment]
@ruflin Made the changes you requested. Ready for your :eyes again, when you get a chance. Thanks!

File: filebeat/module/auditd/log/ingest/pipeline.json (hunk 2 above)
Comment (ycombinator) [reply filebeat/module/auditd/log/ingest/pipeline.json]: Yep, added the `lang` field back in c13108f.

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ycombinator** [comment]
jenkins, test this

**Pull request merged**